tests/openstack/task.yaml

```diff
@@ -0,0 +1,101 @@
+    # fail trying to access through ssh
+    spread openstack:cirros-64: -v -reuse -resend &> task.out || true
+    grep 'cannot find ready marker in console output for .*: timeout reached' task.out
+
+    # Check that the server is discarded during the spread execution
+    for i in $(seq 10); do
+        if [ -z "$(openstack server list)" ]; then
+            break
+        fi
+        sleep 1
+    done
+    test -z "$(openstack server list)"
+
+    # Check the error in case the network does not exist
+    spread openstack:cirros-64-wrong-network: -v -reuse -resend &> task.out || true
+    grep 'cannot find valid network with name "noexist"' task.out
+    test -z "$(openstack server list)"
+
+    # Check the error in case the group does not exist
+    spread openstack:cirros-64-wrong-group: -v -reuse -resend &> task.out || true
+    grep 'cannot find valid group with name "noexist"' task.out
+    test -z "$(openstack server list)"
+
+    # Check the error in case the image does not exist
+    spread openstack:cirros-64-wrong-image: -v -reuse -resend &> task.out || true
+    grep 'cannot find matching image for "noexist"' task.out
+    test -z "$(openstack server list)"
+
+    # Check the error in case the group does not exist
+    spread openstack:cirros-64-wrong-plan: -v -reuse -resend &> task.out  || true
+    grep 'cannot find valid flavor with name "noexist"' task.out
+    test -z "$(openstack server list)"
+
+    # Check errors with environment variables
+    OS_REGION_NAME=wrong spread openstack:cirros-64: -v -reuse -resend &> task.out  || true
+    grep 'cannot authenticate: invalid region "wrong"' task.out
+    OS_PASSWORD=wrong spread openstack:cirros-64: -v -reuse -resend &> task.out  || true
+    grep 'cannot authenticate:.*returned unexpected status: 401' task.out
+    OS_USERNAME=wrong spread openstack:cirros-64: -v -reuse -resend &> task.out  || true
+    grep 'cannot authenticate:.*returned unexpected status: 401' task.out
+    OS_IDENTITY_API_VERSION=2 spread openstack:cirros-64: -v -reuse -resend &> task.out  || true
+    grep -E 'cannot authenticate:.*returned unexpected status: 404' task.out
+
+    # trigger 1 instance and check it can be listed and the garbage collect works
+    test "0" = "$(spread -gc | grep -c "Checking openstack instance")"
+    spread openstack:cirros-64: &>/dev/null &
+
+    for _ in $(seq 10); do
+        if [ -n "$(openstack server list)" ]; then
+            break
+        fi
+        sleep 1
+    done
+
+    # first check that the halt-timeout tag from the instance is used
+    # properties just can be set when the server is in ACTIVE status
+    SERVER_ID="$(openstack server list -f value | head -n1 | awk '{ print $1 }')"
+    test -n "$SERVER_ID"
+    for _ in $(seq 20); do
+        if openstack server show "$SERVER_ID" -f shell | MATCH 'status="ACTIVE"'; then
+            break
+        fi
+        sleep 1
+    done
+    openstack server show "$SERVER_ID" -f shell | MATCH 'status="ACTIVE"'
```

Comment:
So all this test does is check that _some_ server has shown up?  Nothing else at all?